force-app/main/default/classes/Trigger Recipes/MetadataTriggerHandler.cls

```diff
@@ -0,0 +1,131 @@
+         */
+        if (!validateRun()) {
+            return;
+        }
+
+        addToLoopCount();
+
+        /**
+         * End standard validation
+         */
+
+        String objType = getSobjectType();
+        /**
+         * This query finds an ordered list trigger handler classes
+         * to execute. It ignores any classes that are marked as disabled.
+         *
+         * Note: It will exclude any triggerHandler metadata records for which
+         * the user's email address is found in a related disabled_for__mdt
+         * record.
+         *
+         * Admin's and Developers can selectively disable trigger handlers
+         * for all or selected individuals *without* deploying.
+         *
+         */
+        List<Metadata_Driven_Trigger__mdt> tryggers = [
+            SELECT Class__c
+            FROM Metadata_Driven_Trigger__mdt
+            WHERE
+                Object__c = :objType
+                AND Enabled__c = TRUE
+                AND Id NOT IN (
+                    SELECT Metadata_Driven_Trigger__c
+                    FROM Disabled_for__mdt
+                    WHERE UserEmail__c = :UserInfo.getUserEmail()
+                )
+            ORDER BY Execution_Order__c
+        ];
+
+        /**
+         * @description This loop is responsible for instantiating the trigger
+         * handler classes identified by the query above, and then calling the
+         * proper context methods.
+         */
+        for (Metadata_Driven_Trigger__mdt trygger : tryggers) {
+            TriggerHandler handler;
+            try {
+                handler = (TriggerHandler) Type.forName(trygger.class__c)
+                    .newInstance();
+            } catch (Exception e) {
+                continue;
+            }
+
+            switch on context {
+                when BEFORE_INSERT {
+                    handler.beforeInsert();
+                }
+                when BEFORE_UPDATE {
+                    handler.beforeUpdate();
+                }
+                when AFTER_INSERT {
+                    handler.afterInsert();
+                }
+                when AFTER_UPDATE {
+                    handler.afterUpdate();
+                }
+                when BEFORE_DELETE {
+                    handler.beforeDelete();
+                }
+                when AFTER_DELETE {
+                    handler.afterDelete();
+                }
+                when AFTER_UNDELETE {
+                    handler.afterUndelete();
+                }
+            }
+        }
+    }
+
+    /**
+     * @description This determines the active sObject type by describing the first
+     * record in the trigger New / Old list
+     * @return      `String`
+     */
+    private String getSObjectType() {
+        if (Trigger.new != null) {
+            return Trigger.new[0].getSObjectType().getDescribe().getName();
+        } else {
+            return Trigger.old[0].getSObjectType().getDescribe().getName();
```

Comment:
`Trigger.old` can be null in a few contexts (https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_triggers_context_variables.htm)